edd-license-free-download.php

```diff
@@ -260,51 +259,30 @@ public static function process_download() {
 	 * @return bool
 	 */
 	public static function validate_license( $license_key ) {
-		global $wpdb;
-
-		// The edd_license post ID of the license key
-		$license_post_id = $wpdb->get_var(
-		$wpdb->prepare( "SELECT post_id FROM $wpdb->postmeta WHERE meta_key = '_edd_sl_key' AND meta_value = %s", $license_key )
-		 );
-
-		$obj    = edd_software_licensing();
-		$status = $obj->get_license_status( $license_post_id );
-
-		if ( $status != 'expired' ||  $status != 'revoked') {
-			return true;
-		} else {
+		$license = edd_software_licensing()->get_license( $license_key );
+		if ( ! $license ) {
 			return false;
 		}
-	}
+		if ( ! in_array( $license->status, array( 'expired', 'revoked' ), true ) ) {

```

Comment:
I think `revoked` should be `disabled` instead?

Reply:
`revoked` is also a valid status, I believe. It's in SL code. I'm looking to see if there is a definitive status list in there; I would almost prefer to validate against valid statuses instead--a positive comparison instead of a negative?

Reply:
I'm fine with flipping it.

Reply:
Looking through, I think the positive statuses would just be `active` and `inactive`, but I'm not finding what feels like a really definitive list. The admin "views" (which show at the top of the licenses table) seem to indicate the list would be: active, inactive, expired, and disabled. However, there are definitely some examples of code which include the check as we currently have it (expired, disabled, revoked), so I think I can live with the negative after all.